subiquity/server/controllers/tests/test_filesystem.py

```diff
@@ -44,3 +49,54 @@ def test_probe_defaults_and_os(self):
         self.app.opts.use_os_prober = True
         run_coro(self.fsc._probe_once(context=None, restricted=False))
         self.app.prober.get_storage.assert_called_with({'defaults', 'os'})
+
+
+class TestGuided(TestCase):
+    def _guided_direct(self, bootloader, ptable):
+        self.app = make_app()
+        self.app.opts.bootloader = bootloader.value
+        self.controller = FilesystemController(self.app)
+        self.controller.model = make_model(bootloader)
+        self.controller.model._probe_data = {'blockdev': {}}
+        self.d1 = make_disk(self.controller.model, ptable=ptable)
+        self.controller.guided_direct(self.d1)
+
+    def test_guided_direct_UEFI_GPT(self):
+        self._guided_direct(Bootloader.UEFI, 'gpt')
+        [d1p1, d1p2] = self.d1.partitions()
+        self.assertEqual('/boot/efi', d1p1.mount)
+        self.assertEqual('/', d1p2.mount)
+        self.assertEqual(d1p1.size + d1p1.offset, d1p2.offset)
```

Comment:
I've wondered for a while if a helper that checks a disk's partitions for sanity would be good, like here we could use a helper that asserts a disk is full of partitions with no overlaps and no gaps. But it doesn't have to be part of this branch :)

Reply:
That does sound helpful.